File: 401-500/411-420/413-arithmeticSlice/arithmeticSlice.cpp
#include <vector>

using namespace std;

class Solution {
public:
    int numberOfArithmeticSlices(vector<int>& a) {
        int result = 0;
        if (a.size() < 3)
            return result;

        int prev = a[1] - a[0], count = 1;

        for (size_t i = 2; i < a.size(); ++i) {
            int difference = a[i] - a[i-1];
            count = difference == prev ? count + 1 : 1;
            if (count >= 2)
                result += count - 1;

            prev = difference;
        }

        return result;
    }
};

File: 401-500/411-420/413-arithmeticSlice/arithmeticSlice.py
class Solution(object):
    def numberOfArithmeticSlices(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """

        count = 0
        nums.append(float('inf'))
        difference = float('-inf')
        length = len(nums)
        result = 0

        for i in range(1, length):
            diff = nums[i] - nums[i-1]
            if diff == difference:
                count += 1
            else:
                if count >= 2:
                    result += (count - 1) * count / 2

                difference = diff
                count = 1

        return result